src/metabase/api/database.clj

```diff
@@ -1097,22 +1104,36 @@
 
 ;;; ------------------------------------- GET /api/database/:id/schema/:schema ---------------------------------------
 
-(defn- schema-tables-list [db-id schema]
-  (api/read-check Database db-id)
-  (api/check-403 (can-read-schema? db-id schema))
-  (filter mi/can-read? (t2/select Table
-                         :db_id           db-id
-                         :schema          schema
-                         :active          true
-                         ;; a non-nil value means Table is hidden -- see [[metabase.models.table/visibility-types]]
-                         :visibility_type nil
-                         {:order-by [[:display_name :asc]]})))
+(defn- schema-tables-list
+  ([db-id schema include_hidden]
+   (api/read-check Database db-id)
+   (api/check-403 (can-read-schema? db-id schema))
+   (filter mi/can-read? (if include_hidden
+                          (t2/select Table
+                            :db_id db-id
+                            :schema schema
+                            :active true
+                            {:order-by [[:display_name :asc]]})
+                          (t2/select Table
+                            :db_id db-id
+                            :schema schema
+                            :active true
+                            :visibility_type nil
+                            {:order-by [[:display_name :asc]]}))))
+  ([db-id schema]
+   (schema-tables-list db-id schema nil)))
 
 (api/defendpoint GET "/:id/schema/:schema"
   "Returns a list of Tables for the given Database `id` and `schema`"
-  [id schema]
-  {id ms/PositiveInt}
-  (api/check-404 (seq (schema-tables-list id schema))))
+  [id include_hidden include_editable_data_model schema]
+  {id                          ms/PositiveInt
+   include_hidden              [:maybe ms/BooleanString]
+   include_editable_data_model [:maybe ms/BooleanString]}
+  (:tables
+    (db-metadata id
```

Comment:
Another (perhaps cleaner) option here, instead of extending `db-metadata` with another flag, would be to just call `metabase-enterprise.advanced-permissions.common/filter-tables-by-data-model-perms` on the results of `schema-tables-list`.

`filter-tables-by-data-model-perms` could also be converted to a `defenterprise` function so that you don't need to bother with the classloader stuff to call it, like what's being done for `filter-databases-by-data-model-perms`. I don't think I had added `defenterprise` yet when I was working on this code.

Don't worry about this though if this is a time-sensitive escalation, the current approach looks like it works fine as well.